src/groupeddatatable/grouping.jl

```diff
@@ -355,16 +355,23 @@ dt |> groupby(:a) |> [sum, x->mean(dropnull(x))]   # equivalent
 """
 aggregate(d::AbstractDataTable, fs::Function; sort::Bool=false) = aggregate(d, [fs], sort=sort)
 function aggregate{T<:Function}(d::AbstractDataTable, fs::Vector{T}; sort::Bool=false)
```

Comment:
I've been meaning to write a docstring for sort! I always try and use the `by` keyword and get reminded by the error message to use (the undocumented) `cols`
```julia
julia> sort(DataTable(x = rand(10), y = rand(10)), by = :y)
ERROR: ArgumentError: 'by' must be a Function or a vector of Functions. Perhaps you wanted 'cols'.
```

For sorting, these changes *should* be keeping the current behavior. When a user specifies `sort = true`, `groupby` will sort on the grouped columns and then `aggregate` only bothers to sort the remaining columns that weren't sorted during the groupby.

Current code:
```julia
headers = _makeheaders(fs, _setdiff(_names(gd), gd.cols)) # only makes headers for new columns
...
sort && sort!(res, cols=headers) # only sorts new columns
```

I would feel a little more comfortable if we just sorted on all columns at the end rather than two stages of sorting, although I haven't come up with a test case to show that the two-stage sorting might result in the 2nd sort in aggregate improperly re-ordering the 1st sort in groupby